### PR TITLE
[MEAR-294] Removal of provided artifacts from the Class-Path entry of MANIFEST.mf

### PR DESCRIPTION
**Pull request by mabrarov**
[MEAR-294] - Removal of provided artifacts from Class-Path entry of MANIFEST.mf if skinnyWars / skinnyModules options is turned on or if EAR module doesn't contain its dependencies.

### CODE REVIEW

**Review by elharo — COMMENTED** [review]
just some nits. Let me run it through Jenkins

File: src/main/java/org/apache/maven/plugins/ear/AbstractEarMojo.java [hunk 1]
```diff
@@ -317,6 +325,18 @@ public void execute()
         return allJarModules;
     }
 
+    /**
+     * @return The list of {@link #providedJarModules}. This corresponds to provided JAR modules.
```
[the remark this answers is not in the document]
Comment (mabrarov) [reply src/main/java/org/apache/maven/plugins/ear/AbstractEarMojo.java]: Fixed in b9f0931

File: src/test/java/org/apache/maven/plugins/ear/it/EarMojoIT.java [hunk 2]
```diff
@@ -1003,8 +1003,12 @@ public void testProject091()
      * <li>skinnyModules options is turned on</li>
      * </ul>
      * then movement of JARs and modification of manifest Class-Path entry is performed for WAR, SAR, HAR and RAR
-     * modules. Additionally this test ensures that movement of JARs is not performed for modules which
-     * libDirectory property doesn't point to the right module entry containing JAR libraries packaged into module.
+     * modules. Additionally this test ensures that
+     * <ul>
+     * <li>movement of JARs is not performed for modules which libDirectory property doesn't point to correct module
```
[the remark this answers is not in the document]
Comment (mabrarov) [reply src/test/java/org/apache/maven/plugins/ear/it/EarMojoIT.java]: Fixed in b9f0931

**Review by elharo — APPROVED** [review]

**Review by elharo — APPROVED** [review]

**Review by elharo — COMMENTED** [review]
Conflicts with the fix for MEAR-292, which is now committed. Please patch it up, and I'll run Jenkins.

**Comment by mabrarov** [comment]
@elharo,

> Conflicts with the fix for MEAR-292, which is now committed. Please patch it up, and I'll run Jenkins.

Done. Please trigger the Jenkins job.

**Comment by mabrarov** [comment]
What about closing [MEAR-294](https://issues.apache.org/jira/browse/MEAR-294)?

**Comment by jira-importer** [comment]
Resolve #171

**Comment by jira-importer** [comment]
Resolve #171

**Pull request merged**